Add start variable for `run-virus-scan.timer` service

bparry02 opened the pull request:
<!-- GitHub renders PRs such that soft line breaks are treated as hard
line breaks.  In order to make this template render as expected we
therefore have to avoid soft breaks and therefore will offend
markdownlint with long lines.  This is the reason for the markdownline
disable directive just below.

For more details see:
https://github.com/github/markup/issues/1050#issuecomment-294654762 -->
<!-- markdownlint-disable MD013 -->
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

Adds a `clamav_scan_start_timer` role variable that will start the `run-virus-scan.timer` service so the timer will run at the next scheduled time. Defaults to `false` to preserve image build use cases.

Modified the default test scenario to verify that the timer service is not started. Modified the `additional_flags` scenario to override the `run-virus-scan.timer` service state to be started (removed some of the package checks from that scenario but we can add them back if needed).

resolves #103 

## 🧪 Testing ##

```
molecule test -d podman -p fedora43-systemd-amd64 -s default
molecule test -d podman -p fedora43-systemd-amd64 -s additional_flags

molecule test -d podman -p ubuntu-24-systemd-amd64 -s default
molecule test -d podman -p ubuntu-24-systemd-amd64 -s additional_flags
```

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.
- [x] Bump major, minor, patch, pre-release, and/or build versions [as appropriate](https://semver.org/#semantic-versioning-specification-semver) via the `bump_version` script *if* this repository is versioned *and* the changes in this PR [warrant a version bump](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api).

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been -->
<!-- approved. -->

- [x] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [x] Create a release (necessary if and only if the version was bumped).

bparry02 committed:
Add clamav_scan_start_timer variable to start run-virus-scan.timer

bparry02 committed:
Updated additional_flags test verify clamav_scan_start_timer variable

bparry02 commented on tasks/setup_virus_scan.yml:
```diff
@@ -26,4 +26,5 @@
 - name: Enable the systemd timer
   ansible.builtin.systemd_service:
     enabled: true
+    state: "{{ 'started' if clamav_scan_start_timer else omit }}"
```
When false, omitting a value here to preserve the previous role behavior. This could be changed to `else 'stopped'`, which could help when running different test scenarios on the same infra/host (e.g. if additional_flags scenario runs before default, the default scenario will fail because the role will not stop a running timer service).

jsf9k replied:
I think this is good as it is.

jsf9k committed:
Change whitespace to satisfy black pre-commit hook

jsf9k committed:
Bump version from 3.0.1 to 3.1.0

jsf9k committed:
Split test_packages_and_services.py into two files

jsf9k committed:
Adjust/add/remove symlinks

This is necessary now that test_packages_and_services.py has been
split into two files.

jsf9k committed:
Split test_packages_and_services.py in package_install scenario into two files

jsf9k commented on molecule/additional_flags/tests/test_packages_and_services.py:
@bparry02 - With this symlink deleted the "packages" portion of `molecule/default/tests/test_packages_and_services.py` is no longer being tested in the `additional_flags` scenario.  I added commits 63d8015 and 2386330 to remedy this.

jsf9k committed:
Rename test files to avoid module name conflicts

mypy complains when there are two Python files with the same name.

jsf9k committed:
Fix copy-and-paste bug in test

jsf9k commented:
~The failing checks are due to the fact that we are hitting the ClamAV servers hard enough to be temporarily banned.~  The failing Debian 10 check has nothing to do with this PR.  I have created #106 to capture the failure.

dav3r commented on defaults/main.yml:
```diff
@@ -34,6 +34,7 @@ clamav_scan_extra_flags: []
 # format understood by systemd:
 # https://www.freedesktop.org/software/systemd/man/latest/systemd.time.html#Calendar%20Events)
 clamav_scan_frequency: weekly
+clamav_scan_start_timer: false
```
Please move this to maintain alphabetical ordering of the variables.

jsf9k replied:
Please see commit 711d0c0.

dav3r commented on molecule/additional_flags/converge.yml:
```diff
@@ -10,6 +10,7 @@
         name: ansible-role-clamav
       vars:
         clamav_cron_frequency: daily
+        clamav_scan_start_timer: true
```
Please move this to maintain alphabetical ordering of the variables.

jsf9k replied:
Please see commit 711d0c0.

dav3r commented on README.md:
```diff
@@ -34,6 +34,7 @@ None.
 | clamav_scan_exclude_directories | A list of regexes matching directory trees that are to be excluded from scan operations. | `[^/dev, ^/proc, ^/sys, ^/var/spool/clamav]` | No |
 | clamav_scan_extra_flags | Additional flags to pass to clamscan (see clamscan man page for reference).  | `[]` | No |
 | clamav_scan_frequency | The frequency of ClamAV scanning.  Must be a calendar event in [the format understood by systemd](https://www.freedesktop.org/software/systemd/man/latest/systemd.time.html#Calendar%20Events). | `weekly` | No |
+| clamav_scan_start_timer | Whether or not to start the clamav scan timer service. Should be `false` if using this role to build VM images. | `false` | No |
```
Please move this to maintain alphabetical ordering of the variables.

jsf9k replied:
Please see commit 711d0c0.

dav3r reviewed (approved):
LGTM, aside from one nitpick (repeated three times) 👍

jsf9k committed:
Alphabetize role variables

jsf9k commented:
Thank you for the contribution @bparry02!
